core/hooks/name_converter.py

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) 2015 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Hook which chooses an environment file to use based on the current context.
+
+"""
+import os
+import copy
+
+import sgtk
+from tank.util.shotgun import get_sg_connection
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+EDIT_TYPE = "EDIT"
+VALID_EDITS = ["replace", "lower_case", "upper_case", "underscore_to_camelcase"]
+
+class TemplateKeyCustom(HookBaseClass):
+
+    def validate(self, value, validate_transforms, **kwargs):
+        """
+        Test if a value is valid for this key
+
+        :param value: Value to test
+        :param validate_transforms: Bool to enable validation of transforms,
+                                    such as a subset calculation
+        :returns: Bool
+        """
+        # to avoid the validation to fail, if the value is not defined in choices!!
+        return True
```

Comment:
Hrm....this seems bad. Can we talk about why this is necessary?

Reply:
Let me update the stack trace on the ticket, which was the reason I added this hack.